### PR TITLE
Remove hardcoded protocols for octet-stream submission

### PR DESCRIPTION
**Pull request by dplumb94**
Replaces request-promise with built in XMLHTTPRequest functionality.
This fixes a bug where the app could not submit batches or payloads over
TLS. This is because protocol had to be included in the URL for
request-promise, which was hard coded as http. Removing the dependency
allows us to just use a relative URL which will get handled by the proxy

Signed-off-by: Darian Plumb <dplumb@bitwise.io>

### CODE REVIEW

File: examples/gameroom/gameroom-app/src/store/api.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ export async function listGames(circuitID: string): Promise<Game[]> {
 export async function submitPayload(payload: Uint8Array): Promise<void> {
   const options = {
     method: 'POST',
-    url: `http://${window.location.host}/api/submit`,
+    url: `${document.location.protocol}//${window.location.host}/api/submit`,
```
Comment (peterschwarz) [inline examples/gameroom/gameroom-app/src/store/api.ts]: Does this work if you just start the string with `/api/submit` or does it require a complete URL

File: examples/gameroom/gameroom-app/src/store/api.ts [hunk 2]
```diff
@@ -144,43 +175,22 @@ export async function listGames(circuitID: string): Promise<Game[]> {
...
-    headers: {
-      'Content-Type': 'application/octet-stream',
-    },
-  };
-
-  return await rp(options).then((rawBody) => {
+  return await http('POST', `/gamerooms/${circuitID}/batches`, payload, (request: XMLHttpRequest) => {
+    request.setRequestHeader('Content-Type', 'application/octet-stream');
+  }).catch((err) => {
+    throw new Error(err);
```
Comment (peterschwarz) [inline examples/gameroom/gameroom-app/src/store/api.ts]: This is redundant.

File: examples/gameroom/gameroom-app/src/store/api.ts [hunk 3]
```diff
@@ -144,43 +175,22 @@ export async function listGames(circuitID: string): Promise<Game[]> {
...
-    },
-  };
-
-  await rp(options).then((body) => {
-    return;
-  })
-  .catch((err) => {
-    console.error(err.message);
-    throw new Error('Failed to send request. Contact administrator for help.');
+  await http('POST', '/submit', payload, (request: XMLHttpRequest) => {
```
Comment (peterschwarz) [inline examples/gameroom/gameroom-app/src/store/api.ts]: Do you need to parse the JSON Response on this one?

Comment (dplumb94) [reply examples/gameroom/gameroom-app/src/store/api.ts]: If anything goes wrong at this level, I don't think it should get displayed to the user. I am console.erroring the error in `http()` for debugging.

File: examples/gameroom/gameroom-app/src/store/api.ts (hunk 2 above)
Comment (dplumb94) [reply examples/gameroom/gameroom-app/src/store/api.ts]: This is required to send the error back to the component to invoke the toast alert

**Pull request merged**